main.go

```diff
@@ -1204,6 +1171,43 @@ func _main(ctx context.Context) error {
 		return err
 	}
 
+	// Retrieve blockchain deployment updates and add them to the agendas db.
+	// activeChain.Deployments contains a list of all agendas supported in the
+	// current environment.
+	if err = agendasInstance.CheckAgendasUpdates(activeChain.Deployments); err != nil {
+		return fmt.Errorf("updating agendas db failed: %v", err)
+	}
+
+	// Piparser should run updates only after the initial sync
+	if !cfg.DisablePiParser {
+		// Initiate the piparser handler here.
+		pgDB.StartPiparserHandler()
+
+		// Retrieve newly added proposals and add them to the proposals db(storm).
+		// Proposal db update is made asynchronously to ensure that the system works
+		// even when the Politeia API endpoint set is down.
+		go func() {
+			if err := proposalsInstance.CheckProposalsUpdates(); err != nil {
+				log.Errorf("updating proposals db failed: %v", err)
+			}
+		}()
+
+		// It initiates the updates fetch process for the proposal votes data after
+		// sync for the other tables is complete. It is only run if the system is on
+		// full mode. An error in fetching the updates should not stop the system
+		// functionality since it could be attributed to the external systems used.
+		log.Info("Running updates retrieval for Politeia's Proposals. Please wait...")
+
+		// Fetch updates for Politiea's Proposal history(votes) data via the parser.
+		commitsCount, err := pgDB.PiProposalsHistory()
+		if err != nil {
+			log.Errorf("pgDB.PiProposalsHistory failed : %v", err)
+		} else {
+			log.Infof("%d politeia's proposal (auxiliary db) commits were processed",
+				commitsCount)
+		}
+	}
+
```

Comment:
Nothing that requires a non-trivial amount of time to execute should come between `ensureSync` and the startup of the notifiers.  These prop and agenda updates can take time, so they should be moved.  They can be after the notifiers start, but I think they can also come way earlier, any time after the table indexes are built since that's the only reason for delaying these updates.